frontend/src/metabase/components/DownloadButton.jsx

```diff
@@ -23,7 +23,8 @@ function colorForType(type) {
 }
 
 const retrieveFilename = ({ res, type }) => {
-  const contentDisposition = res.headers.get("Content-Disposition") || "";
+  const contentDisposition =
+    decodeURIComponent(res.headers.get("Content-Disposition")) || "";
```

Comment:
If the `Content-Disposition` header is missing and `get` returns `null`, `decode` will coerce it to string (`"null"`) and potentially cause unexpected behavior. I believe `||` check should be done on the header getter and not on the `decodeURIComponent` result, for example:

```js
const contentDispositionHeader = res.headers.get("Content-Disposition") || "";
const contentDisposition = decodeURIComponent(contentDispositionHeader);
```